.github/workflows/pr-test-lint.yml

```diff
@@ -172,6 +172,29 @@ jobs:
           fail-on-alert: true
           save-data-file: false
           alert-threshold: '150%'
+
+  go-auto-approve:
+    runs-on: ubuntu-latest
+    needs: [golangci-lint, go-test, go-test-integration, go-bench, go-mod]
+    # For now, we only auto approve and merge provider release PRs created by mondoo-tools.
+    # We have to check the commit author, because the PR is created by "github-actions[bot]"
+    # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#startswith
+    if: ${{ startsWith(github.ref, 'refs/heads/version/providers_update_') && github.event.commits[0].author.username == 'mondoo-tools' }}
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      # figure out the PR for this commit
+      - uses: cloudposse-github-actions/get-pr@v1.0.1
+        id: pr
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          filterOutClosed: true
+          filterOutDraft: true
+      - name: Approve a PR
+        uses: fastify/github-action-merge-dependabot@v3
```

Comment:
Just letting you both know @benr @philipbalinov that we are using this new GH action.

Reply:
We also have this already in cnspec and packer-plugin repo